src/core/producer/stage.cpp

```diff
@@ -69,19 +98,60 @@ struct stage::impl : public std::enable_shared_from_this<impl>
                 for (auto& t : tweens_)
                     t.second.tick(1);
 
+                // build a map of layers that are sourced from route producers
+                std::map<int, std::pair<int, int>> routed_layers;
                 for (auto& p : layers_) {
-                    auto& layer = p.second;
-                    auto& tween = tweens_[p.first];
+                    auto producer = std::move(p.second.foreground());
+                    if (0 == producer->name().compare(L"route")) {
+                        std::wstring         routeSrc(producer->print());
+                        boost::wsmatch       what;
+                        static boost::wregex expr(L"route\\[(?<CHANNEL>\\d+)(\\/(?<LAYER>\\d+))?\\]",
```

Comment:
Could we do something with extending/changing the types instead of relying on regex?
Perhaps like the dynamic_pointer_cast done a few lines below. Or will that have a significant performance impact from all the failed casts?

Reply:
Agreed, done.